Source/Core/Gen4/Generators/WildGenerator4.cpp

```diff
@@ -254,6 +254,7 @@ std::vector<WildState4> WildGenerator4::generateMethodK(u32 seed, const Encounte
     u8 buffer = 0;
     u8 thresh = 0;
     u16 rate = encounterArea.getEncounterRate();
+    u8 loops = 1;
```

Comment:
I am still not sure I like how this is handled. It seems very invasive to handle one special case. I think the core logic can be extracted to a function and the different encounters use this function as necessary